### PR TITLE
Add multiexporterexporter.

### PR DESCRIPTION
**Pull request by kovrus**
Resolves https://github.com/grafana/otel-squad/issues/40

### CODE REVIEW

File: components/exporter/multiexporterexporter/exporter.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	case exporter.Logs:
+		ld, ok := data.(plog.Logs)
+		if !ok {
+			return fmt.Errorf("invalid data type %q for the logs exporter", reflect.TypeOf(data))
+		}
+		return v.ConsumeLogs(ctx, ld)
+	case exporter.Traces:
+		td, ok := data.(ptrace.Traces)
+		if !ok {
+			return fmt.Errorf("invalid data type %q for the traces traces", reflect.TypeOf(data))
```
Comment (jpkrohling) [inline components/exporter/multiexporterexporter/exporter.go:59]:
```suggestion
			return fmt.Errorf("invalid data type %q for the traces exporter", reflect.TypeOf(data))
```

File: components/exporter/multiexporterexporter/exporter.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		}
+		return v.ConsumeTraces(ctx, td)
+	case exporter.Metrics:
+		md, ok := data.(pmetric.Metrics)
+		if !ok {
+			return fmt.Errorf("invalid data type %q for the metrics exporter", reflect.TypeOf(data))
+		}
+		return v.ConsumeMetrics(ctx, md)
+	default:
+		return errors.New("")
```
Comment (jpkrohling) [inline components/exporter/multiexporterexporter/exporter.go:69]: TODO

File: components/exporter/multiexporterexporter/exporter.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	}
+}
+
+func (m *multiExporter) start(ctx context.Context, host component.Host) (err error) {
+	for _, e := range m.exporters {
+		exp, ok := e.(component.Component)
+		if !ok {
+			return fmt.Errorf("registered exporter is not a component")
```
Comment (jpkrohling) [inline components/exporter/multiexporterexporter/exporter.go:77]: State which exporter you are complaining about.

File: components/exporter/multiexporterexporter/exporter_test.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+				assert.ErrorContains(t, err, tt.error)
+				return
+			}
+		})
+	}
+}
+
+func TestRetrieveFromContext(t *testing.T) {
+	md := collectorclient.NewMetadata(map[string][]string{
+		"X-Scope-Orgid": {"123"},
```
Comment (jpkrohling) [inline components/exporter/multiexporterexporter/exporter_test.go:162]: Can we cross-test this with the case normalization? Like, use `OrgID` here and `Orgid` on line 168 (leaving a comment stating that this is on purpose)

File: components/exporter/multiexporterexporter/factory.go [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+	exp := newMultiExporter(exporters, c.MappingKey, params.Logger)
+	return exporterhelper.NewTracesExporter(
+		ctx,
+		params,
+		cfg,
+		func(ctx context.Context, td ptrace.Traces) error {
+			return exp.consume(ctx, td)
+		},
+		exporterhelper.WithStart(exp.start),
```
Comment (jpkrohling) [inline components/exporter/multiexporterexporter/factory.go:73]: Doesn't the underlying exporter need to be shut down as well?

**Review by jpkrohling — COMMENTED** [review]
I'm missing some metrics for the component, but the general direction looks good.

**Comment by kovrus** [comment]
closing for now since there is no work planned in this direction. I'll keep it in the `multiexporterexporter` branch if anyone is interested.

**Pull request closed without merging**